pkg/controller/gc_controller.go

```diff
@@ -84,9 +86,63 @@ func NewGCController(
 		restoreClient:       restoreClient,
 		logger:              logger,
 	}
-}
 
-var _ Interface = &gcController{}
+	backupInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(old interface{}, new interface{}) {
+				var (
+					backup = new.(*api.Backup)
+					log    = logger.WithField("backup", kube.NamespaceAndName(backup))
+					idx    = -1
+				)
+
+				// we're only interested in backups that have a deletionTimestamp and at
+				// least one finalizer.
+				if backup.DeletionTimestamp == nil || len(backup.Finalizers) == 0 {
+					return
+				}
+				log.Debugf("Backup has finalizers %s", backup.Finalizers)
+
+				for x, finalizer := range backup.Finalizers {
```

Comment:
I'd extract this to a function that just returns a bool. No need for idx.

Reply:
And then maybe make another helper that returns a new slice w/the finalizer removed. See e.g. https://github.com/kubernetes/kubernetes/blob/722cb7a758d4678e7ed5c4f09ded99d6bc5b0b4c/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/helpers.go#L109-L118